optimizer_wrapper.rb

```diff
@@ -679,31 +694,38 @@ def self.compute_result_total_dimensions_and_round_route_stats(result)
   end
 
   def self.compute_route_waiting_times(route)
-    seen = 1
-    previous_end =
-      if route[:activities].first[:type] == 'depot'
-        route[:activities].first[:begin_time]
-      else
-        route[:activities].first[:end_time]
-      end
+    setup_types = %w[service pickup delivery]
+    previous_end = route[:activities].first[:end_time] || route[:activities].first[:begin_time]
     route[:activities].first[:waiting_time] = 0
 
     first_service_seen = true
-    while seen < route[:activities].size
-      considered_setup =
-        if route[:activities][seen][:type] == 'rest'
-          0
-        elsif first_service_seen || route[:activities][seen][:travel_time].positive?
-          route[:activities][seen][:detail][:setup_duration] || 0
-        else
-          0
+    loc_index = nil
+    consumed_travel_time = 0
+    route[:activities][1..-1].each.with_index{ |act, index|
+      used_travel_time = 0
+      if act[:type] == 'rest'
+        if loc_index.nil?
+          loc_index = 1 + index + route[:activities][1 + index..-1].index{ |a| a[:type] != 'rest' }
```

Comment:
Here, are we trying to handle the case where there are multiple pauses one after the other?